py/server/deephaven/experimental/iceberg.py

```diff
@@ -116,47 +276,41 @@ def tables(self, namespace: str) -> Table:
             a table containing the tables in the provided namespace.
         """
 
-        if namespace is not None:
-            return Table(self.j_object.listTablesAsTable(namespace))
-        return Table(self.j_object.listTablesAsTable())
+        return Table(self.j_object.tables(namespace))
 
-    def snapshots(self, table_identifier: str) -> Table:
+    def load_table(self, table_identifier: str) -> IcebergTable:
         """
-        Returns information on the snapshots of the specified table as a Deephaven table.
+        Load the table from the catalog using the provided instructions.
 
         Args:
-            table_identifier (str): the table from which to list snapshots.
+            table_identifier (str): the table to read.
 
         Returns:
-            a table containing the snapshot information.
+            Table: the table read from the catalog.
         """
 
-        return self.j_object.listSnapshotsAsTable(table_identifier)
+        return IcebergTableAdapter(self.j_object.loadTable(table_identifier))
 
-    def read_table(self, table_identifier: str, instructions: Optional[IcebergInstructions] = None, snapshot_id: Optional[int] = None) -> Table:
+    def snapshots(self, table_identifier: str) -> Table:
         """
-        Reads the table from the catalog using the provided instructions. Optionally, a snapshot id can be provided to
-        read a specific snapshot of the table.
+        NOTE: deprecated, use `load_table(table_identifier).snapshots()` instead.
```

Comment:
Should this be deleted?

Reply:
If it is in `experimental`, I think we need to be aggressive about deleting deprecated things.  I would only leave it if we have a paying client that will be grouchy if there isn't a stop gap for a release.

Reply:
Also, if it is deprecated, it should be deprecated in the python way.  @jmao-denver can provide details.